src/main/java/org/junit/Assert.java

```diff
@@ -955,4 +955,48 @@ public static void assertEquals(Object[] expecteds, Object[] actuals) {
             Matcher<? super T> matcher) {
         MatcherAssert.assertThat(reason, actual, matcher);
     }
+
+    /**
+     * This interface facilitates the use of expectThrows from Java 8. It allows method references
+     * to void methods (that declare checked exceptions) to be passed directly into expectThrows
+     * without wrapping. It is not meant to be implemented directly.
+     *
+     * @since 4.13
+     */
+    public interface ThrowingRunnable {
+        void run() throws Throwable;
+    }
+
+    /**
+     * Asserts that {@code runnable} throws an exception of type {@code throwableClass} when
+     * executed. If it does, the exception object is returned. If it does not throw an exception, an
+     * {@link AssertionError} is thrown. If it throws the wrong type of exception, an {@code
+     * AssertionError} is thrown describing the mismatch; the exception that was actually thrown can
+     * be obtained by calling {@link AssertionError#getCause}.
+     *
+     * @param throwableClass the expected type of the exception
+     * @param runnable       A function that is expected to throw an exception when executed
+     * @return The exception thrown by {@code runnable}
+     * @since 4.13
+     */
+    public static <T extends Throwable> T expectThrows(Class<T> throwableClass, ThrowingRunnable runnable) {
+        try {
+            runnable.run();
+        } catch (Throwable t) {
+            if (throwableClass.isInstance(t)) {
+                return throwableClass.cast(t);
```

Comment:
@rschmitt,

Can I ask a favor?  I'd like the chance to have a conversation with @marcphilipp and some others about how we should make choices given the upcoming plans for JUnit Lambda.  These people have invested considerable time away from their paid day jobs in order to look into these very questions, and I think it's fair to them to give them time to agree on a plan, and fair to you and other contributors to see a plan before you invest time into a particular enhancement.

Given that time to work on JUnit is, for all the maintainers, time that must be gleaned from outside of paid work and family, this conversation could take a while.  Can you give us two weeks?

If that's really too long, then you might want to try publishing this in a separate fork or repo for user feedback, because I'm afraid that JUnit has never been (nor needed to be) an organization that makes decisions much more quickly than that.

Feel free to drop me a note individually if more than two weeks goes by without an update here.

Fair enough?